### PR TITLE
refactor(api): extract semantic generator + LLM enrichment into one shared engine (#3233)

### LINKED ISSUE #3233: refactor(api): extract semantic generator + LLM enrichment into shared lib (CLI + wizard call one engine)
The semantic-layer generator and its LLM enrichment run in the wrong places to be reused. Mechanical YAML generation lives in `packages/api/src/lib/profiler.ts`, but the **enrichment** pass lives in CLI-only `packages/cli/bin/enrich.ts` — so the web wizard (`wizard.ts`) can only ever emit mechanical output, and "what `atlas init --enrich` produces" can drift from "what the wizard produces." Consolidate both behind one shared engine that the CLI and the API routes call identically.

## What to build

- Extract the **mechanical generator** + the **LLM enrichment** logic into a shared module under `packages/api/src/lib/semantic/` (e.g. a `generate/` + `enrich/` surface).
- The engine exposes: (a) profile → mechanical entity/catalog/glossary/metric YAML; (b) enrich a given table's YAML via an LLM round-trip that may run read-only DB queries (distribution/sample) to ground its output.
- Both front-ends call the **same** engine: `atlas init` (CLI) and the `wizard.ts` routes (web). No logic forks between them.
- Respect the layering rules (CLAUDE.md): `lib/` sits **above** `api/routes/`; `lib/` must **not** import from `api/routes/`. Mock all exports in tests; no top-level singleton mutation.

This is a pure refactor enabler — no user-visible behavior change in this slice. It unblocks group-namespace generation and the two-phase generate UI.

## Key files

- `packages/cli/bin/enrich.ts` — enrichment logic to move into shared lib
- `packages/api/src/lib/profiler.ts` — `generateEntityYAML` / `generateCatalogYAML` / `generateGlossaryYAML` / `generateMetricYAML` / `analyzeTableProfiles`
- `packages/cli/src/commands/init.ts` — CLI caller (rewire to shared engine)
- `packages/api/src/api/routes/wizard.ts` — web caller (rewire to shared engine)
- `docs/design/semantic-onboarding.md` (§ F)

## Acceptance criteria

- [ ] Mechanical generation + enrichment live in `packages/api/src/lib/semantic/`; `lib/` does not import from `api/routes/`.
- [ ] `atlas init` and `atlas init --enrich` produce identical output to before (snapshot/behavior parity).
- [ ] `wizard.ts` `/generate` produces identical mechanical output to before via the shared engine.
- [ ] Enrichment is callable from both the CLI and the API path with the same signature/result shape.
- [ ] Tests cover the shared engine directly (mechanical + enrichment) with mocked LLM + connection mocks; CLI/wizard callers covered.

## Dependencies

None — can start immediately.

### PR DESCRIPTION
**Pull request by msywulak**
Closes #3233.

## What & why

Mechanical YAML generation lived in `packages/api/src/lib/profiler.ts`, but the **LLM enrichment** pass lived CLI-only in `packages/cli/bin/enrich.ts`. So the web wizard could only ever emit *mechanical* output, and "what `atlas init --enrich` produces" could drift from "what the wizard produces." This consolidates both behind **one shared engine** under `packages/api/src/lib/semantic/`, called identically by the CLI and the web wizard.

This is a **pure refactor** — no user-visible behavior change. Parity is the bar; the value is consolidation (it unblocks #3234 group-scoping and the two-phase generate UI).

## The move

| Surface | From | To |
|---|---|---|
| Mechanical generator (`analyzeTableProfiles`, `generate{Entity,Catalog,Metric,Glossary}YAML` + heuristics) | `lib/profiler.ts` | `lib/semantic/generate/` (`analyze.ts` + `yaml.ts`) |
| LLM enrichment (`enrichSemanticLayer` + per-table `enrich{Entity,Glossary,Metric}`) | `cli/bin/enrich.ts` (deleted) | `lib/semantic/enrich/` |

- **`lib/profiler.ts` re-exports the generator surface** for backward compatibility, so every existing consumer — `cli/bin/atlas.ts`, `cli/lib/diff.ts`, and ~10 profiler test files — is **untouched**. profiler.ts drops ~880 lines.
- **Both front-ends call the same engine, no logic forks:** `wizard.ts` and `atlas init` import generators from `@atlas/api/lib/semantic/generate`; `atlas init --enrich` lazily imports `@atlas/api/lib/semantic/enrich`.
- Enrichment's `TableProfile` now comes from canonical `@useatlas/types` (was the CLI's `bin/atlas.js`). Same `enrichSemanticLayer(profiles, { semanticDir })` signature — **callable from both the CLI and the API path**.

## Layering (CLAUDE.md)

The engine lives in `lib/`, which sits **above** `api/routes/`; it does not import from `api/routes/`. The wizard route's helper logic was not needed by the engine, so nothing moved the wrong way.

The engine is reached **only via its own package-export paths** — it is deliberately **not** re-exported from the `semantic/index.ts` barrel. `enrich` pulls in the AI SDK + `providers`, and the barrel is imported broadly (sync, startup); an eager re-export forced that heavy graph onto every barrel consumer and broke providers-mocking tests (caught locally — `startup.test.ts`). New `package.json` `exports` entries map the two directory modules (mirroring the existing `./lib/semantic/expert` pattern).

`wizard.ts` `connectionId` scoping is left as-is — group-namespace scoping is #3234's job.

## Tests / parity

- **Mechanical parity, unchanged suites:** the existing profiler suites (138 API + 97 CLI assertions) and the wizard + audit tests pass **with zero test-file edits** through the re-export facade — the strongest parity signal.
- **New direct-path coverage** of the shared engine: a `generate-engine` smoke (imports from the new path) and `enrich-engine` (the previously-**untested** enrichment) — mocked LLM (`ai.generateText` + `providers.getModel`) over a real temp-dir of seed YAML, asserting the read→merge→write behavior, the no-matching-profile skip, and the unparseable-response safe path. No connection mocks: this slice's enrichment makes no DB calls (DB-grounded enrichment is a later slice).

## CI

`/ci` green locally: lint, type, full test (changed surface 100% green — API affected 10/10, CLI 29/29; 3 unrelated admin tests flaked under full-suite concurrency and pass in isolation), syncpack, template-drift, security-headers, railway-watch, schema-drift, oauth-helper, test-discipline, twenty-resolver, published-symbols.

## Acceptance criteria
- [x] Mechanical generation + enrichment live in `packages/api/src/lib/semantic/`; `lib/` does not import from `api/routes/`.
- [x] `atlas init` / `atlas init --enrich` produce identical output (existing suites green unchanged).
- [x] `wizard.ts` `/generate` produces identical mechanical output via the shared engine.
- [x] Enrichment is callable from both the CLI and the API path with the same signature.
- [x] Tests cover the shared engine directly (mechanical + enrichment) with a mocked LLM; CLI/wizard callers covered.

---
<a href="https://app.blacksmith.sh/AtlasDevHQ/codesmith/atlas/pr/3242"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783307415&installation_id=135337507&pr_number=3242&repository=AtlasDevHQ%2Fatlas&return_to=https%3A%2F%2Fgithub.com%2FAtlasDevHQ%2Fatlas%2Fpull%2F3242&signature=3eef907126334d4270c77b257a9e191c11a524a841bcea44ed65a25ca089a43d"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

* **New Features**
  * Stable, public semantic generator and enrichment engines; CLI and web wizard now produce consistent, richer YAML (FK inference, view detection, enum handling, denormalization/abandonment flags).

* **Tests**
  * Added comprehensive generation and enrichment tests covering YAML output, merge semantics, fallback behavior, and edge cases.

* **Chores**
  * Reorganized semantic/profiling code into modular export surfaces for clearer onboarding and init workflows.

### CODE REVIEW

File: packages/api/src/lib/semantic/__tests__/enrich-engine.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+const mockGenerateText = mock(async ({ prompt }: GenArgs) => {
+  let text = "no usable yaml here";
+  if (prompt.includes("enriching a semantic layer YAML file")) text = ENTITY_RESPONSE;
+  else if (prompt.includes("building a business glossary")) text = GLOSSARY_RESPONSE;
+  else if (prompt.includes("enriching metric definitions")) text = METRIC_RESPONSE;
+  return { text, usage: { promptTokens: 11, completionTokens: 22, totalTokens: 33 } };
+});
+
+mock.module("ai", () => ({ generateText: mockGenerateText }));
+mock.module("@atlas/api/lib/providers", () => ({ getModel: () => ({ modelId: "test-model" }) }));
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/lib/semantic/__tests__/enrich-engine.test.ts]: Fixed in c233ba8a — both the `ai` and `@atlas/api/lib/providers` mocks now spread the real module (`...aiActual` / `...providersActual`) and override only `generateText`/`getModel`, so any other export (`gateway`, `PROVIDER_KEY_MAP`, etc.) stays intact. Thanks!

File: packages/api/src/lib/semantic/generate/yaml.ts [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+        label: `${col.name.replace(/_/g, " ")} by ${enumCol.name}`,
+        description: `${col.name} broken down by ${enumCol.name}.`,
+        type: "atomic",
+        sql: `SELECT ${enumCol.name}, ${aggFunc}(${col.name}) as ${aggLabel}_${col.name}, COUNT(*) as count\nFROM ${qualifiedTable}\nGROUP BY ${enumCol.name}\nORDER BY ${aggLabel}_${col.name} DESC`,
+      });
+    }
+  }
+
+  return yaml.dump({ metrics }, { lineWidth: 120, noRefs: true });
+}
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/lib/semantic/generate/yaml.ts:611]: Good catch, but intentionally left as-is in this PR. #3242 is a pure, parity-bound refactor — this `generateMetricYAML` block moved byte-for-byte from `lib/profiler.ts`, where it already had the `schema !== "public"` check while `generateEntityYAML` uses `&& schema !== "main"`. Applying the fix here would change generated DuckDB/SQLite (`main` schema) metric SQL and break the parity bar. Filed separately as #3244 to fix as a deliberate behavior change. @coderabbitai
